src/App.jsx

```diff
@@ -14,33 +16,86 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_FIELD_ALPHABETICALLY = 'alphabetically';
+const SORT_FIELD_LENGTH = 'length';
+const SORT_FIELD_RESET = 'reset';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+export const App = () => {
+  const [viewGoods, setViewGoods] = useState(goodsFromServer);
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  const sortByAlphabetic = () => {
+    setSortField(SORT_FIELD_ALPHABETICALLY);
+    const sortedGoods = [...goodsFromServer].sort((good1, good2) =>
+      good1.localeCompare(good2));
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+    setViewGoods(isReversed ? sortedGoods.reverse() : sortedGoods);
+  };
+
+  const sortByLength = () => {
+    setSortField(SORT_FIELD_LENGTH);
+    const sortedGoods = [...goodsFromServer].sort(
+      (good1, good2) => good1.length - good2.length,
+    );
+
+    setViewGoods(isReversed ? sortedGoods.reverse() : sortedGoods);
+  };
+
+  const sortReverse = () => {
+    setIsReversed(prev => !prev);
+    setViewGoods([...viewGoods].reverse());
+  };
+
+  const sortReset = () => {
+    setSortField(SORT_FIELD_RESET);
```

Comment:
Setting the `sortField` to `SORT_FIELD_RESET` is not necessary for the functionality, as it doesn't affect the button styling or any other logic. Consider removing this line if it's not needed for future extensions.